llms/hf_llm/generate.py

```diff
@@ -0,0 +1,82 @@
+# Copyright © 2023 Apple Inc.
+
+import argparse
+import time
+
+import mlx.core as mx
+import models
+import transformers
+
+
+def generate(
+    model: models.Model,
+    tokenizer: transformers.AutoTokenizer,
+    prompt: str,
+    max_tokens: int,
+    temp: float = 0.0,
+):
+    prompt = tokenizer(args.prompt, return_tensors="np", return_attention_mask=False,)[
+        "input_ids"
+    ][0]
+    prompt = mx.array(prompt)
+
+    tic = time.time()
+    tokens = []
+    skip = 0
+    for token, n in zip(
+        models.generate(prompt, model, args.temp),
+        range(args.max_tokens),
+    ):
+        if token == tokenizer.eos_token_id:
+            break
+
+        if n == 0:
```

Comment:
That should be after `token.item()` to properly measure the timing of the prompt processing.

Reply:
Actually the `==` above forces the eval. Sorry it's very implicit...

Reply:
Oh nice! Completely overlooked that 🤦‍♂️